assembly/src/tokens/stream.rs

```diff
@@ -21,15 +28,34 @@ impl<'a> TokenStream<'a> {
         if source.is_empty() {
             return Err(AssemblyError::empty_source());
         }
+        let mut doc_comments = BTreeMap::new();
+
+        let mut tokens = Vec::new();
+        let mut comment = Comment(String::new());
 
-        let tokens = source
-            .lines()
-            // Tokenize and remove comments
-            .flat_map(|line| {
-                line.split_whitespace()
+        let lines = source.lines();
+        for line in lines {
+            let line = line.trim();
+            if line.starts_with(DOC_COMMENT_PREFIX) {
+                comment.append_line(line);
+            } else if line.starts_with(FUNC_PREFIX) {
+                doc_comments.insert(tokens.len(), comment.get_comment());
+                comment.clear();
+                let mut line_tokens = line
+                    .split_whitespace()
+                    .take_while(|&token| !token.starts_with('#'))
+                    .collect::<Vec<_>>();
+                tokens.append(&mut line_tokens);
```

Comment:
I am not sure if we need to treat `FUNC_PREFIX` separately from others. I think we should be able to get rid of this, and just put something like this in the `else` clause:
```Rust
if !comment.is_empty() {
    doc_comments.insert(tokens.len(), comment.get_comment());
    comment.clear();
}
```
Although, I'd probably have a single method on the `Comment` struct which does both. For example: `comment.take_content()`.